pkg/R/utils.R

```diff
@@ -164,11 +176,11 @@ reserialize <- function(rdd) {
   if (!inherits(rdd, "RDD")) {
     stop("Argument 'rdd' is not an RDD type.")
   }
-  if (rdd@env$serialized) {
-    return(rdd)
-  } else {
+  if (rdd@env$serialized == "string") {
```

Comment:
I think we should do this for `!= "byte"` to be very safe -- I maybe wrong though.
@sun-rui @hlin09  -- Do you have a better idea of whether `reserialize` is required for the new Row-based serialization ?